src/java.net.http/share/classes/jdk/internal/net/http/common/Deadline.java

```diff
@@ -0,0 +1,223 @@
+     * The amount is typically {@link Duration} but may be any other type implementing
+     * the {@link TemporalAmount} interface.
+     * <p>
+     * This instance is immutable and unaffected by this method call.
+     *
+     * @param amountToAdd  the amount to add, not null
+     * @return a {@code Deadline} based on this deadline with the addition made, not null
+     * @throws DateTimeException if the addition cannot be made
+     * @throws ArithmeticException if numeric overflow occurs
+     */
+    public Deadline plus(TemporalAmount amountToAdd) {
+        return Deadline.of(deadline.plus(amountToAdd));
+    }
+
+    /**
+     * Calculates the amount of time until another deadline in terms of the specified unit.
+     * <p>
+     * This calculates the amount of time between two {@code Deadline}
+     * objects in terms of a single {@code TemporalUnit}.
+     * The start and end points are {@code this} and the specified deadline.
+     * The result will be negative if the end is before the start.
+     * The calculation returns a whole number, representing the number of
+     * complete units between the two deadlines.
+     * <p>
+     * This instance is immutable and unaffected by this method call.
+     *
+     * @param endExclusive  the end deadline, exclusive
+     * @param unit  the unit to measure the amount in, not null
+     * @return the amount of time between this deadline and the end deadline
+     * @throws DateTimeException if the amount cannot be calculated
+     * @throws UnsupportedTemporalTypeException if the unit is not supported
+     * @throws ArithmeticException if numeric overflow occurs
+     */
+    public long until(Deadline endExclusive, TemporalUnit unit) {
+        return deadline.until(endExclusive.deadline, unit);
+    }
+
+    /**
+     * Checks if this deadline is after the specified deadline.
+     * <p>
+     * The comparison is based on the time-line position of the deadlines.
+     *
+     * @param otherDeadline  the other deadline to compare to, not null
+     * @return true if this deadline is after the specified deadline
+     * @throws NullPointerException if otherDeadline is null
+     */
+    public boolean isAfter(Deadline otherDeadline) {
+        return compareTo(otherDeadline) > 0;
+    }
+
+    /**
+     * Checks if this deadline is before the specified deadline.
+     * <p>
+     * The comparison is based on the time-line position of the deadines.
+     *
+     * @param otherDeadline  the other deadine to compare to, not null
+     * @return true if this deadline is before the specified deadine
+     * @throws NullPointerException if otherDeadline is null
+     */
+    public boolean isBefore(Deadline otherDeadline) {
+        return compareTo(otherDeadline) < 0;
+    }
+
+    @Override
+    public int compareTo(Deadline o) {
+        return deadline.compareTo(o.deadline);
+    }
+
+    @Override
+    public String toString() {
+        return deadline.toString();
```

Comment:
I'm not sure about using `Instant.toString` here; if we ever log a `Deadline` instance along with the current time and the clock skews, we will need to read the code along with the logs to figure out which date represents a deadline and which one represents an instant.
I guess the problem would also go away if we initialized the first NanoSource with something other than Instant.now().

Reply:
I can modify the toString to be 
```
  return "Deadline(" + deadline.toString() + ")";
```
if you think it's better.

But I do believe that using a fake instant for the origin of the NanoSource would be way more confusing.

Reply:
We can also avoid using that method. I guess we can always deal with it later if it proves to be a problem.